### PR TITLE
Added: AzureRM implementation

### PR DESCRIPTION
**Pull request by xescugc**
Just supporting `virtual_machine`(linux) and `linux_virtual_machine` with the `size`.

Related #3

### CODE REVIEW

File: e2e/azurerm_ingestion_test.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	backend := mysql.NewBackend(db)
+	ingester, err := azurerm.NewIngester(ctx, azurerm.VirtualMachines.String(), region, azurerm.WithIngestionFilter(azurerm.MinimalFilter), azurerm.WithEndpoint(ts.URL))
+	require.NoError(t, err)
+
+	err = costestimation.IngestPricing(ctx, backend, ingester)
+	require.NoError(t, err)
+
+	allProds, err := backend.Products().Filter(ctx, &product.Filter{Provider: util.StringPtr(azurerm.ProviderName), Service: util.StringPtr(azurerm.VirtualMachines.String())})
+	require.NoError(t, err)
+	fmt.Println(len(allProds))
```
Comment (xlr-8) [inline e2e/azurerm_ingestion_test.go:44]: ?

File: testdata/azurerm/plan.json [hunk 2]
```diff
@@ -0,0 +1,1384 @@
...
+				"env": {
+					"description": "Cycloid environment name."
+				},
+				"project": {
+					"description": "Cycloid project name."
+				}
+			}
+		}
+	}
+}
```
Comment (xlr-8) [inline testdata/azurerm/plan.json:1384]: I didn't see any sensitive data, I assume it got cleaned up? Asking just to double check

File: testdata/azurerm/stack-compute/vm.tf [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+#  direction                   = "Outbound"
+#  access                      = "Allow"
+#  protocol                    = "Tcp"         # Possible values: "Tcp", "Udp", "Icmp", "Esp", "Ah" or "*"
+#  source_port_range           = "*"
+#  destination_port_range      = "*"
+#  source_address_prefix       = "*"
+#  destination_address_prefix  = "*"
+#  resource_group_name         = module.vm.resource_group_name
+#  network_security_group_name = module.vm.network_security_group_name
+#}
```
Comment (xlr-8) [inline testdata/azurerm/stack-compute/vm.tf:219]: Seems like it can be removed

**Review by xlr-8 — CHANGES_REQUESTED** [review]
Awesome! Minor comments :+1:

File: testdata/azurerm/plan.json (hunk 2 above)
Comment (xescugc) [reply testdata/azurerm/plan.json:1384]: Yes double checked.

**Comment by xescugc** [comment]
Done :)

**Review by xlr-8 — APPROVED** [review]
RS

**Comment by xescugc** [comment]
Done

**Pull request merged**